geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py

```diff
@@ -0,0 +1,152 @@
+                                self.pseudo3d_params.u_cell_size,
+                                self.pseudo3d_params.v_cell_size,
+                            ],
+                            self.pseudo3d_params.depth_core,
+                            [self.pseudo3d_params.horizontal_padding] * 2
+                            + [self.pseudo3d_params.vertical_padding, 1],
+                            self.pseudo3d_params.expansion_factor,
+                        )[0]
+
+                        iter_workspace.remove_entity(receiver_entity.current_electrodes)
+                        iter_workspace.remove_entity(receiver_entity)
+
+                        xyz_out = mesh.centroids
+                        model_uids = deepcopy(models)
+                        for name, model in models.items():
+                            if model is None:
+                                continue
+                            elif isinstance(model, Data):
+                                model_values = weighted_average(
+                                    xyz_in, xyz_out, [model.values], n=1
+                                )[0]
+                            else:
+                                model_values = model * np.ones(len(xyz_out))
+
+                            model_object = mesh.add_data(
+                                {name: {"values": model_values}}
+                            )
+                            model_uids[name] = model_object.uid
+
+                        for key in ifile.data:
+                            param = getattr(self.pseudo3d_params, key, None)
+                            if hasattr(param, "uid"):
+                                if not isinstance(param, Data):
+                                    param.copy(
+                                        parent=iter_workspace, copy_children=True
+                                    )
+
+                            ifile.data[key] = param
+
+                        ifile.data.update(
+                            dict(
+                                lookup[uuid],
+                                **{
+                                    "title": ifile.data["title"].replace("batch ", ""),
+                                    "inversion_type": ifile.data[
+                                        "inversion_type"
+                                    ].replace("pseudo 3d", "2d"),
+                                    "geoh5": iter_workspace,
+                                    "mesh": mesh,
+                                    "line_id": trial["line_id"],
+                                },
+                                **model_uids,
+                            )
+                        )
+
+                    ifile.name = f"{uuid}.ui.json"
+                    ifile.path = self.working_directory
+                    ifile.write_ui_json()
+                    lookup[uuid]["status"] = "written"
+                else:
+                    lookup[uuid]["status"] = status
```

Comment:
I don't think this does anything since you just pop and re-insert. Consider the simpler logic suggested above.

Reply:
done